src/test/java/org/mastodon/mamut/io/exporter/graphml/GraphMLUtilsTest.java

```diff
@@ -0,0 +1,105 @@
+import org.mastodon.collection.RefSet;
+import org.mastodon.mamut.feature.branch.exampleGraph.ExampleGraph4;
+import org.mastodon.mamut.model.branch.BranchLink;
+import org.mastodon.mamut.model.branch.BranchSpot;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class GraphMLUtilsTest
+{
+
+	private ExampleGraph4 graph4;
+
+	@Before
+	public void setUp()
+	{
+		graph4 = new ExampleGraph4();
+	}
+
+	@Test
+	public void testExportAllTracks() throws IOException
+	{
+		File tempDirectory = Files.createTempDirectory( "prefix" ).toFile();
+		String prefix = "test";
+		GraphMLUtils.exportAllTracks( graph4.getModel().getBranchGraph(), tempDirectory, prefix );
+
+		File[] files = tempDirectory.listFiles();
+		assertNotNull( files );
+		Set< String > actual = new HashSet<>( Arrays.asList( Arrays.stream( files ).map( File::getName ).toArray( String[]::new ) ) );
+
+		Set< String > expected = new HashSet<>();
+		expected.add( "test-0.graphml" );
+		expected.add( "test-4.graphml" );
+		assertEquals( expected, actual );
+	}
+
+	@Test
+	public void testExportBranches() throws IOException
+	{
+		RefSet< BranchSpot > branchSpots = RefCollections.createRefSet( graph4.getModel().getBranchGraph().vertices() );
+		branchSpots.add( graph4.branchSpotA );
+		branchSpots.add( graph4.branchSpotB );
+		branchSpots.add( graph4.branchSpotC );
+		branchSpots.add( graph4.branchSpotD );
+		branchSpots.add( graph4.branchSpotE );
+		RefSet< BranchLink > branchLinks = RefCollections.createRefSet( graph4.getModel().getBranchGraph().edges() );
+		branchLinks.add( graph4.branchLink0 );
+		branchLinks.add( graph4.branchLink1 );
+		branchLinks.add( graph4.branchLink2 );
+
+		File tempFile = Files.createTempFile( "graphml-test", "" ).toFile();
+		GraphMLUtils.exportBranches( branchSpots, branchLinks, tempFile );
+
+		// read file to string - java 8 style
+		String actualContent = new String( Files.readAllBytes( tempFile.toPath() ) );
+		actualContent = actualContent.replaceAll( "[\\t\\n\\r\\s]", "" );
+
+		// expected content
+		String expectedContent =
+				"<?xml version=\"1.0\" encoding=\"UTF-8\"?><graphml xmlns=\"http://graphml.graphdrawing.org/xmlns\" xsi:schemaLocation=\"http://graphml.graphdrawing.org/xmlns http://graphml.graphdrawing.org/xmlns/1.0/graphml.xsd\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\">\n"
+						+ "    <key id=\"vertex_label_key\" for=\"node\" attr.name=\"branchName\" attr.type=\"string\"/>\n"
+						+ "    <key id=\"key0\" for=\"node\" attr.name=\"duration\" attr.type=\"int\"/>\n"
+						+ "    <graph edgedefault=\"directed\">\n"
+						+ "        <node id=\"4\">\n"
+						+ "            <data key=\"vertex_label_key\">6</data>\n"
+						+ "            <data key=\"key0\">1</data>\n"
+						+ "        </node>\n"
```

Comment:
It would be very nice if the name of the first spot of a branch would be used as the "id".
But it's not trivial to achieve because spot names are not unique, so would need to find a way to make the id unique.

Reply:
I am not sure, if I am getting your suggestion.

The current behaviour is like this:

![grafik](https://github.com/mastodon-sc/mastodon-deep-lineage/assets/10515534/9299cea3-7c89-4c6b-b8d6-cfe5c09606dc)

![grafik](https://github.com/mastodon-sc/mastodon-deep-lineage/assets/10515534/e91b7f2e-5505-4da1-aa61-19f0a95ff4ab)

Reply:
As discussed in person. This was just a misunderstanding.